std/datetime/systime.d

```diff
@@ -9031,22 +9069,64 @@ private:
       +/
     @property void adjTime(long adjTime) @safe nothrow scope
     {
-        _stdTime = _timezone.tzToUTC(adjTime);
+        _stdTime = timezoneHack.tzToUTC(adjTime);
     }
 
 
-    // Commented out due to bug http://d.puremagic.com/issues/show_bug.cgi?id=5058
-    /+
-    invariant()
+    final class InitTimeZone : TimeZone
+    {
+    public:
+
+        static immutable(InitTimeZone) opCall() @safe pure nothrow @nogc { return _initTimeZone; }
+
+        @property override bool hasDST() @safe const nothrow @nogc { return false; }
+
+        override bool dstInEffect(long stdTime) @safe const nothrow @nogc { return false; }
+
+        override long utcToTZ(long stdTime) @safe const nothrow @nogc { return 0; }
+
+        override long tzToUTC(long adjTime) @safe const nothrow @nogc { return 0; }
+
+        override Duration utcOffsetAt(long stdTime) @safe const nothrow @nogc { return Duration.zero; }
+
+    private:
+
+        this() @safe immutable pure
+        {
+            super("SysTime.init's timezone", "SysTime.init's timezone", "SysTime.init's timezone");
```

Comment:
If there's a chance that this string is ever going to be displayed in an application, I suggest instead something like `"uninitialized timezone"`.

Reply:
Why? That seems less clear to me. The problem isn't that the timezone was uninitialized (especially if the compiler bug is finally fixed). The problem is that `SysTime.init` is being used instead of giving the variable an actual, useful value.

Reply:
>Why?

Because "SysTime.init's timezone" it's meaningless to the end user of the application. If it's only for debugging purposes and `TimeZone.getTimeZone` that's fine. Otherwise it needs to be written for humans.

Reply:
Are there any legitimate uses of `SysTime.init`?

Reply:
Not really. It indicates that the `SysTime` hasn't been initialized yet, and you could use it if you needed to subtract from midnight, January 1st, 1 AD, but realistically, it only gets used when someone is operating on a `SysTime` that hasn't been properly initialized, and occasionally, such values get printed (which currently segfaults), so having that not blow up would be nice, but really, all `SysTime.init` is there for is to be the default value, not to be actually used for anything beyond that.